src/openforms/registrations/contrib/objects_api/validators.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import warnings
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+from openforms.contrib.zgw.clients.catalogi import Catalogus
+
+from .client import get_catalogi_client
+from .models import ObjectsAPIGroupConfig
+
+
+def validate_catalogue_reference(
+    config: ObjectsAPIGroupConfig,
+) -> Catalogus | None:
+    domain = config.catalogue_domain
+    rsin = config.catalogue_rsin
+
+    if not domain and not rsin:
+        return
+    assert domain and rsin, "Domain and RSIN need to both be provided"
```

Comment:
If I edit an `ObjectsAPIGroupConfig` and only provide one or the other, I do not get a validationerror but an assertionerror (500):

```
AssertionError at /admin/registrations_objects_api/objectsapigroupconfig/1/change/

Domain and RSIN need to both be provided
```

Reply:
ah nice catch - probably because the check constraint validation never happens and thus the associated error message is not displayed. That's unfortunate :(

I'll come up with a solution and test.